doc/semantic_version.md

```diff
@@ -0,0 +1,43 @@
+# WAMR uses semantic versioning
+
+WAMR uses the _semantic versioning_ to replace current _date versioning_ system.
+
+There are three parts in the new version string:
+
+- _major_. WAMR only changes _major_ when APIs are modified in an incompatible way.
+  APIs includes: `wasm_export.h`, `wasm_c_api.h`, _sections in AOT files_, and so on.
```

Comment:
> APIs includes: `wasm_export.h`, `wasm_c_api.h`

does this mean ABI changes are not relevant?

> _sections in AOT files_

what do you mean?

Reply:
- Our APIs do not just include headers but also AOT internal structures, like section headers, offsets, fields, and so on.
- Both ABI and API matter.

Reply:
> Both ABI and API matter.

it's confusing because all legacy releases below share the major 1, while there were actually abi-breaking changes.

Reply:
my suggestion is
* simply start from 1.0.0
* just say "pre-1.0.0 releases don't follow semver convention"